File: solved/Problem_002.py
# By considering the terms in the Fibonacci sequence whose values do not exceed four million, find the sum of the even-valued terms.

def main():
    total = 0
    (a, b) = (1, 2)
    while b < 4000000:
        (a, b) = (b, a+b)
        if a%2 == 0:
            total += a

    return total

if __name__ == "__main__":

    print main()

File: java/Problem_002.java
public class Problem_002 {
	public static void main (String[] args) {
		int a = 1;
		int b = 2;
		int total = 0;
		while (b <= 4000000) {
			if (b%2 == 0) {
				total += b;
			}
			b = b + a; 
			a = b - a;
		}
		System.out.println(total);
	}
}
